ui/page/root/home_page.go

```diff
@@ -203,7 +212,8 @@ func (hp *HomePage) HandleUserInteractions() {
 		case values.String(values.StrWallets):
 			pg = hp.walletSelectorPage
 		case values.String(values.StrTrade):
-			pg = NewTradePage(hp.Load)
+			hp.ensureDEXIsReady()
+			pg = NewTradePage(hp.Load, hp.dexCore)
```

Comment:
`hp.ensureDEXIsReady` doesn't really ensure dex is ready, dexCore might still be nil after that method returns. Should check here and NOT naviagate to the dex page if dexCore could not be initialized. Show a toast message or something instead.

Reply:
> and NOT naviagate to the dex page if dexCore could not be initialized. Show a toast message or something instead.

We have both the CEX page and DEX page under the Trade page. This means we have to display the trade page but handle a possible nil dex client in the DEX page. I've done this in my lastest review changes.